_studio/mfx_lib/encode_hw/hevc/agnostic/base/hevcehw_base_legacy.cpp

```diff
@@ -539,6 +542,30 @@ void Legacy::SetInherited(ParamInheritance& par)
         INHERIT_OPT(VideoFullRange);
         INHERIT_OPT(ColourDescriptionPresent);
     });
+
+    par.m_ebInheritDefault[MFX_EXTBUFF_AVC_TEMPORAL_LAYERS].emplace_back(
+        [](const mfxVideoParam& /*parInit*/
+            , const mfxExtBuffer* pSrc
+            , const mfxVideoParam& /*parReset*/
+            , mfxExtBuffer* pDst)
+    {
+        INIT_EB(mfxExtAvcTemporalLayers);
+        // In HEVC spec, nuh_temporal_id_plus1 minus 1 specifies a temporal identifier for the NAL unit
+        // nuh_temporal_id_plus1 is 3 bits only. Then the valid range for layers number is [0,6]
+        bool bEmptyLayer = true;
+        for (mfxU32 i = 0; i < 7; i++)
+        {
+            if (ebReset.Layer[i].Scale > 0) bEmptyLayer = false;
+        }
+        if (bEmptyLayer)
+        {
+            for (mfxU32 i = 0; i < 7; i++)
+            {
+                INHERIT_OPT(Layer[i].Scale);
```

Comment:
Do we really need to inherit init values on reset?

Reply:
In case when we reset other parameter, if Layer.Scale in Init Par is 1,2,0,0,0,0 for example, and not inherited to reset par. Then, we can't see temporal layers during reset, is that expected?

Reply:
@JeremyShang , do you plan to address the comments?

Reply:
@JeremyShang  ?

Reply:
Hi @dmitryermilov, sorry for late response. I have offline talk with them, it is concluded that it is not a bug to set AvcTemporalLayers in every reset, so I think I can close this PR. Please leave comments if you still have any. Thanks!